Watchman.Discord/Areas/Protection/Controllers/MuteUserController.cs

```diff
@@ -13,32 +19,46 @@ public class MuteUserController : IController
         private readonly MutingService _mutingService;
         private readonly UnmutingService _unmutingService;
         private readonly UsersService _usersService;
+        private readonly MessagesServiceFactory _messagesServiceFactory;
 
-        public MuteUserController(MutingService mutingService, UnmutingService unmutingService, UsersService usersService)
+        public MuteUserController(MutingService mutingService, UnmutingService unmutingService, UsersService usersService, MessagesServiceFactory messagesServiceFactory)
         {
             this._mutingService = mutingService;
             this._unmutingService = unmutingService;
             this._usersService = usersService;
+            this._messagesServiceFactory = messagesServiceFactory;
         }
 
-        [DiscordCommand("mute")]
         [AdminCommand]
-        public async Task MuteUser(DiscordRequest request, Contexts contexts)
+        public async Task MuteUser(MuteCommand command, Contexts contexts)
         {
-            var requestParser = new MuteRequestParser(request, this._usersService, contexts);
-            var userToMute = requestParser.GetUser();
-            var muteEvent = requestParser.GetMuteEvent(userToMute.Id, contexts, request);
+            var requestParser = new MuteRequestParser(this._usersService, contexts);
+            var userToMute = await requestParser.GetUser(command.Mention);
+            MuteEvent? muteEvent = null;
```

Comment:
`MuteEvent?`
ale przecież to typ referencyjny
to po co ma być nullable? co mu to ma dać?

Reply:
Zrobiłem tak, bo IDE mi pokazuje ostrzeżenie (konwertowanie literału o wartości null lub możliwej wartości null na nienullowany typ) i zasugerowało mi abym tak zrobił.

Reply:
myśle że warto to wyłączyć
bo za dużo korzyści z tego nie widze, a trzeba stosować takie obejścia problemu